Library/Homebrew/unpack_strategy.rb

```diff
@@ -88,24 +116,31 @@ def self.detect(path, prioritize_extension: false, type: nil, ref_type: nil, ref
     strategy.new(path, ref_type:, ref:, merge_xattrs:)
   end
 
-  attr_reader :path, :merge_xattrs
+  sig { returns(Pathname) }
+  attr_reader :path
 
-  def initialize(path, ref_type: nil, ref: nil, merge_xattrs: nil)
-    @path = Pathname(path).expand_path
+  sig { returns(T::Boolean) }
+  attr_reader :merge_xattrs
+
+  sig {
+    params(path: T.any(String, Pathname), ref_type: T.nilable(String), ref: T.nilable(String),
+           merge_xattrs: T::Boolean).void
+  }
+  def initialize(path, ref_type: nil, ref: nil, merge_xattrs: false)
+    @path = T.let(Pathname(path).expand_path, Pathname)
     @ref_type = ref_type
     @ref = ref
     @merge_xattrs = merge_xattrs
   end
 
-  abstract!
-  sig { abstract.params(unpack_dir: Pathname, basename: Pathname, verbose: T::Boolean).returns(T.untyped) }
+  sig { abstract.params(unpack_dir: Pathname, basename: Pathname, verbose: T::Boolean).void }
```

Comment:
All the overrides of this should also be updated to be `void`. (I'm a little surprised that isn't a type error.) 

I think this `sig` should also be marked `override` (though i easily get lost in how OS extensions work): https://github.com/Homebrew/brew/blob/master/Library/Homebrew/extend/os/mac/unpack_strategy/zip.rb#L11

Reply:
Maybe `T.untyped` is fine here then?

Reply:
Eh, `T.untyped` is a bit cheating to get to `strict`, IMO.

But, realistically, it's fine to punt on this here, as it's already an ambitious PR.

Reply:
> Eh, `T.untyped` is a bit cheating to get to `strict`, IMO.

For everything except a Hash value: yeh, I agree, particularly on returns. Sorry 😭

Reply:
Fixed!

Reply:
I tried marking that `extend/os/mac/unpack_strategy` sig as override but Sorbet said it was unnecessary.